### PR TITLE
[feat]: add new samples java interop - service discovery

### PR DESCRIPTION
**Pull request by Yoda-wu**
i finish the samples with java-go interop service discovery both service level and interface level
related issue: https://github.com/apache/dubbo-go-samples/issues/698

### CODE REVIEW

**Review by DMwangnima — COMMENTED** [review]
Nice job! BTW, pls fix the CI problem and add this test case to CI.

**Comment by DMwangnima** [comment]
For dubbo java part, cc @FoghostCn .

**Review by justxuewei — COMMENTED** [review]
Please check your code:

- Comments start with a space, for example:

```
//Comment goes here (bad case
// Comment goes here (good case)
```

- Remove unused code instead of leaving it commented out.

**Comment by justxuewei** [comment]
Btw, I'd like to suggest you formating your java code, e.g. [google-java-format](https://github.com/google/google-java-format). What formatting tool is used in Dubbo?  /cc @chickenlj

**Comment by Yoda-wu** [comment]
i will fix it soon

File: service_discovery/service/go-server/server.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+type GreetTripleServer struct {
+}
+
+func (svr *GreetTripleServer) Greet(ctx context.Context, req *greet.GreetRequest) (*greet.GreetResponse, error) {
+	resp := &greet.GreetResponse{Greeting: req.Name}
+	return resp, nil
+}
+
+func (s *GreetTripleServer) MethodMapper() map[string]string {
```
[the remark this answers is not in the document]
Comment (Yoda-wu) [reply service_discovery/service/go-server/server.go:25]: sorry, it turn out that java client can still invoke the go server. i made a mistake. The reason why i  add `MethodMapper` is https://cn.dubbo.apache.org/zh-cn/overview/mannual/golang-sdk/tutorial/develop/interflow/call_java/#22-java-client---go-server
and i referenced the wrong part (Part2 Hessian2 serialize)

File: service_discovery/service/java-client/src/main/java/org/apache/dubbo/samples/Main.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import org.apache.dubbo.samples.proto.GreetRequest;
+import org.apache.dubbo.samples.proto.GreetResponse;
+import org.apache.dubbo.samples.proto.GreetService;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+
+public class Main {
+    private static GreetService greetService;
```
Comment (FoghostCn) [inline service_discovery/service/java-client/src/main/java/org/apache/dubbo/samples/Main.java:29]: pls make greetService as method private

**Comment by FoghostCn** [comment]
The generated code does not need to be submitted, just configure the maven idl plugin

**Comment by FoghostCn** [comment]
pls  add a async usage in java client 
```java
CompletableFuture<GreetResponse> future = greetService.greetAsync(req);
GreetResponse greet = future.get();
```

File: service_discovery/service/java-client/pom.xml [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  ~  See the License for the specific language governing permissions and
+  ~  limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.apache.dubbo</groupId>
+    <artifactId>samples</artifactId>
```
Comment (FoghostCn) [inline service_discovery/service/java-client/pom.xml:24]: Please use a more meaningful module name

File: service_discovery/interface/java-client/pom.xml [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  ~  See the License for the specific language governing permissions and
+  ~  limitations under the License.
+  -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.apache.dubbo</groupId>
+    <artifactId>samples</artifactId>
```
Comment (FoghostCn) [inline service_discovery/interface/java-client/pom.xml:24]: Please use a more meaningful module name

**Review by FoghostCn — CHANGES_REQUESTED** [review]

**Comment by Yoda-wu** [comment]
ok, i will fix it soon

**Comment by FoghostCn** [comment]
> Btw, I'd like to suggest you formatting your java code, e.g. [google-java-format](https://github.com/google/google-java-format). What formatting tool is used in Dubbo? /cc @chickenlj

spotless plugin is used in dubbo-java
```
mvn spotless:apply
```

**Pull request merged**